### PR TITLE
Added router test to check if openshift-monitoring is able to pull haproxy metrics

### PR DESCRIPTION
**Pull request by pravisankar**
- Also fixed router test cases related to metrics

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/sig-network-edge PTAL

**Comment by pravisankar** [comment]
Depends on https://github.com/openshift/origin/pull/22002

File: test/extended/router/metrics.go [hunk 1]
```diff
@@ -306,3 +332,73 @@ func getAuthenticatedURLViaPod(ns, execPodName, url, user, pass string) (string,
...
+
+	username, ok1 := secret.Data[userKey]
+	password, ok2 := secret.Data[passKey]
+	if !ok1 || !ok2 {
+		return "", "", fmt.Errorf("secret '%s/%s' does not contain key %q or %q", ns, secretName, userKey, passKey)
+	}
+	return string(username), string(password), nil
+}
+
+func findMetricsBearerToken(oc *exutil.CLI) (string, error) {
```
Comment (ironcladlou) [inline test/extended/router/metrics.go]: for future discussion: what about running the test pod using an SA with the right permissions?

**Comment by ironcladlou** [comment]
I think refactoring can be done later, we need this now

/lgtm

**Comment by ironcladlou** [comment]
This is consistently failing:

    openshift-tests [Conformance][Area:Networking][Feature:Router] The HAProxy router should serve routes that were created from an ingress [Suite:openshift/conformance/parallel/minimal] 3m38s

We also have other tests failing sometimes where the router is failing health checks despite lack of errors in the router logs:

https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-4.0/4057/#openshift-tests-conformanceareanetworkingfeaturerouter-the-haproxy-router-should-override-the-route-host-for-overridden-domains-with-a-custom-value-suiteopenshiftconformanceparallelminimal

We're going to have to dig in and understand why the router is returning 503 intermittently...

**Pull request merged**